### PR TITLE
Enable use of ddrgen from OMR

### PR DESCRIPTION
**Pull request by keithc-ca**
* add new UMA flag 'opt_useOmrDdr'
* add rule to make *.i from *.m4 for ddrgen
* adjust modules and configure OMR accordingly
* omit work based on legacy ddr generation tools
* run omr/getmacros to discover constants
* run omr/ddrgen to produce blob and superset

Depends on https://github.com/eclipse/omr/pull/2364.

### CODE REVIEW

File: runtime/ddr/module.xml [hunk 1]
```diff
@@ -1,351 +1,429 @@
...
+			<dependency name="j9jvmti"/>
+			<dependency name="j9prt"/>
+			<dependency name="j9shr"/>
+			<dependency name="j9thr"/>
+			<dependency name="j9trc"/>
+			<dependency name="j9vm"/>
+			<!--
+			We actually only need one of the following, but
+			the version we're building doesn't appear to be
+			availabue to UMA, so we'll require them all.
```
Comment (pshipton) [inline runtime/ddr/module.xml]: typo availabue

**Comment by pshipton** [comment]
@dmitripivkine fyi

**Comment by keithc-ca** [comment]
Fixed type and squashed.

**Comment by pshipton** [comment]
jenkins compile

**Comment by pshipton** [comment]
The JDK9 failures are related to finding freemarker (fixed by #1361), and the ci build passed. The JDK8 compile on aix isn't finished yet, but the OpenJ9 part is. I believe this is good to merge.

**Pull request merged**